core/src/main/java/org/apache/calcite/sql2rel/RelStructuredTypeFlattener.java

```diff
@@ -534,10 +535,21 @@ public void rewriteRel(LogicalProject rel) {
     RelNode newInput = getNewForOldRel(rel.getInput());
     List<RexNode> newProjects = Pair.left(flattenedExpList);
     List<String> newNames = Pair.right(flattenedExpList);
-    final RelNode newRel = relBuilder.push(newInput)
+    RelNode renamedProject =  relBuilder.push(newInput)
         .projectNamed(newProjects, newNames, true)
```

Comment:
I would prefer not to touch `RelBuilder#projectNamed` within this PR, otherwise it might require revising of all the places where `projectNamed` is called